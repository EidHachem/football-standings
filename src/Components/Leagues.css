.leagues {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  background-color: #3e61a3;
}

.league {
  display: flex;
  padding: 0.2em;
  min-height: 10em;
}

.league:nth-child(4n + 2) {
  background-color: #677eb0;
}

.league:nth-child(4n + 3) {
  background-color: #677eb0;
}

.upper-league {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: space-between;
}

.league-name {
  align-self: flex-end;
  color: #fff;
  margin: 0.5em 0;
  min-width: 4em;
  font-size: 1.1rem;
}

.image {
  max-width: 90%;
}

.league-btn {
  border: none;
  background-color: #4369b2;
  cursor: pointer;
}

h2 {
  color: #fff;
  margin: 0;
  font-size: 1.1rem;
  padding: 0.2em 0;
}
