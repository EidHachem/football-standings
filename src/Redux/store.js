import { configureStore } from '@reduxjs/toolkit';
import leaguesSlice from './leaguesSlice/leaguesSlice';

const store = configureStore({
  reducer: {
    leagues: leaguesSlice,
  },
});

export default store;
