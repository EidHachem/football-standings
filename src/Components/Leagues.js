import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchLeagues } from '../Redux/leaguesSlice/leaguesSlice';
import { BiRightArrowCircle } from 'react-icons/bi';
import './Leagues.css';

const Leagues = () => {
  const leagues = useSelector((state) => state.leagues);

  const dispatch = useDispatch();

  useEffect(() => {
    if (!leagues.length) dispatch(fetchLeagues());
  }, [dispatch, leagues.length]);

  return (
    <>
      <div className="grid-container">
        <div>
          <h2>International leagues</h2>
        </div>
        <div className="leagues">
          {leagues.map((league) => (
            <div className="league" key={league.id}>
              <div>
                <img src={league.logo} alt={league.name} className="image" />
              </div>
              <div className="upper-league">
                <BiRightArrowCircle
                  className="league-btn"
                  style={{
                    width: '25px',
                    height: '25px',
                    color: '#fff',
                    background: 'transparent',
                  }}
                />
                <p className="league-name">{league.name}</p>
              </div>
            </div>
          ))}
        </div>
      </div>
    </>
  );
};

export default Leagues;
