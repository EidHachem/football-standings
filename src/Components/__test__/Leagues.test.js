import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
// import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/store';
import Leagues from '../Leagues';

describe('Leagues test', () => {
  it('renders leagues', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Leagues />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
