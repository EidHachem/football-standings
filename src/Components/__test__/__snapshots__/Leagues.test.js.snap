// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leagues test renders leagues 1`] = `
<div
  className="grid-container"
>
  <div>
    <h2>
      International leagues
    </h2>
  </div>
  <div
    className="leagues"
  />
</div>
`;
